src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -466,6 +480,44 @@ export class StudyViewPageStore
             )
         );
 
+        this.reactionDisposers.push(
+            reaction(
+                () => [this.filtersProx, this.hesitateUpdate],
+                () => {
+                    if (!this.hesitateUpdate) {
+                        this.filters = this.filtersProx;
+                    }
+                    this.filterUpdateTime = performance.now();
+                },
+                {
+                    equals: comparer.structural,
+                }
+            )
+        );
+
+        /**
+         * Submit filters when submit mode changes
+         * or when user clicks submit button
+         */
+        this.reactionDisposers.push(
+            reaction(
+                () => [this.hesitateUpdate, this.filterSubmitTime],
+                () => {
+                    _.forIn(this.hesitantPillStore, (value, key) => {
+                        const onDeleteCallback = value.onDeleteCallback;
+                        if (onDeleteCallback) {
+                            onDeleteCallback();
+                            delete this.submittedPillStore[key];
+                        } else {
+                            this.submittedPillStore[key] = value;
+                        }
+                    });
+                    this.hesitantPillStore = {};
+                    this.filters = this.filtersProx;
```

Comment:
maybe it's this here?  this is going to update filters every time hesitateUpdate is changed, even when filters are exactly the same.  just a hunch.